sdk/schemaregistry/azure-schemaregistry-avroencoder/migration_guide.md

```diff
@@ -0,0 +1,194 @@
+
+```python
+with serializer:
+    dict_data_ben = {"name": "Ben", "favorite_number": 7, "favorite_color": "red"}
+    payload_ben = serializer.serialize(value=dict_data_ben, schema=SCHEMA_STRING)
+    print(payload_ben)  # prints bytes b'<record format indicator><schema ID><Avro-encoded data>``
+```
+
+Using the `AvroEncoder`:
+
+```python
+with encoder:
+    dict_content = {"name": "Ben", "favorite_number": 7, "favorite_color": "red"}
+    message_content_ben = encoder.encode(content=dict_content, schema=definition)
+    print(message_content_ben)  # prints {"content": <Avro-encoded content>, "content_type": "<Avro MIME type>+<Schema ID>"}
+```
+
+#### Decode
+
+The positional parameter `value` has been renamed to `content`. If `value` parameter name was specified when calling `deserialize`, it must be updated to `content`.
+
+Using the `AvroSerializer`:
+
+```python
+with serializer:
+    decoded_value = serializer.deserialize(value=encoded_payload)
+```
+
+Using the `AvroEncoder`:
+
+```python
+with encoder:
+    decoded_content = encoder.decode(value=encoded_content)
+```
+
+### Content type
+
+Previously, `encode` returned an Avro-encoded content, with the binary encoded schema ID and record format indicator values prepended to it. Now, `encode` will return a TypedDict of {`content`: `<serialized payload>`, `content_type`: `<Avro MIME type>`+`<schema ID>`} by default.
+
+#### Backward compatibility
+In order to decode content that was encoded (or "serialized") by the `AvroSerializer`:
+
+1. Install `azure-schemaregistry-avroencoder` v1.0.0b2, which automatically detects the preamble.
+
+```pwsh
+pip install azure-schemaregistry-avroencoder==1.0.0b2
+```
+
+2. Receive and decode events from Event Hubs that were "serialized" by the `AvroSerializer`.
+
+```python
+import os
+from azure.eventhub import EventHubConsumerClient
+from azure.schemaregistry import SchemaRegistryClient
+from azure.schemaregistry.encoder.avroencoder import AvroEncoder
+from azure.identity import DefaultAzureCredential
+
+token_credential = DefaultAzureCredential()
+fully_qualified_namespace = os.environ['SCHEMAREGISTRY_FULLY_QUALIFIED_NAMESPACE']
+group_name = os.environ['SCHEMAREGISTRY_GROUP']
+eventhub_connection_str = os.environ['EVENT_HUB_CONN_STR']
+eventhub_name = os.environ['EVENT_HUB_NAME']
+
+schema_registry_client = SchemaRegistryClient(fully_qualified_namespace, token_credential)
+avro_encoder = AvroEncoder(client=schema_registry_client, group_name=group_name)
+
+eventhub_consumer = EventHubConsumerClient.from_connection_string(
+    conn_str=eventhub_connection_str,
+    consumer_group='$Default',
+    eventhub_name=eventhub_name,
+)
+
+def on_event(partition_context, event):
+    decoded_content = avro_encoder.decode(event)
+
+with eventhub_consumer, avro_encoder:
+    eventhub_consumer.receive(on_event=on_event, starting_position="-1")
+```
+
+This backward compatibility was removed starting `azure-schemaregistry-avroencoder` v1.0.0b3.
```

Comment:
You might want to put this towards the top that backwards compatibility was removed after beta 3.